test/functional/tests/io_class/test_io_class_lvm_on_cas_dss.py

```diff
@@ -0,0 +1,151 @@
+        description: Validate the ability of CAS to cache IO class when CAS device is used as LVM.
+        pass_criteria:
+          - Create CAS device and LVM on top of it successfully.
+          - Loading IO class configuration successfully.
+          - Running FIO for file size from IO class from 11 to 21 successfully.
+          - Increasing proper statistics as expected.
+    """
+    with TestRun.step(f"Create CAS device."):
+
+        cache_dev, core_dev = prepare_devices()
+
+        cache = casadm.start_cache(cache_dev[0], cache_mode=CacheMode.WB, force=True)
+        core = cache.add_core(core_dev[0])
+
+    with TestRun.step("Create LVM on CAS device."):
+        lvm_filters = ["a/.*/", "r|/dev/sd*|", "r|/dev/hd*|", "r|/dev/xvd*|", "r/disk/", "r/block/",
+                       "r|/dev/nvme*|"]
+        pv_num = 1
+        vg_num = 1
+        lv_num = 1
+        cache_num = 1
+        cas_dev_num = 1
+
+        config = LvmConfiguration(lvm_filters, pv_num, vg_num, lv_num, cache_num, cas_dev_num)
+
+        lvms = Lvm.create_specific_lvm_configuration(core, config)
+        lvm = lvms[0]
+
+    with TestRun.step("Create filesystem for LVM and mount it."):
+        lvm.create_filesystem(Filesystem.ext4)
+        lvm.mount(mountpoint)
+
+    with TestRun.step("Load IO class config."):
+        cache.load_io_class("/etc/opencas/ioclass-config.csv")
+
+    with TestRun.step("Run fio for file size from IO class from 11 to 21 "
+                      "and check that correct statistics increase."):
+        csv = casadm.list_io_classes(cache.cache_id, OutputFormat.csv).stdout
+        io_classes = IoClass.csv_to_list(csv)
+        file_size = 2048
+        for io_class in io_classes:
+            if io_class.id < 11 or io_class.id > 21:
+                continue
+            print(f"IO Class ID: {io_class.id}, class name: {io_class.rule}")
+            cache.reset_counters()
+
+            print("Running FIO with verification on LVM.")
+            run_fio(file_size)
+
+            print("Checking statistics.")
+            check_statistics(cache, io_classes, io_class.id)
+
+            if file_size < 256 * 1024 * 1024:
+                file_size *= 4
+            else:
+                file_size = 1100 * 1024 * 1024  # > 1 GiB
+
+            fs_utils.remove(io_target)
+
+    with TestRun.step("Remove LVMs."):
+        TestRun.executor.run(f"umount {mountpoint}")
+        Lvm.remove_all()
+
+
+def prepare_devices():
+    cache_device = TestRun.disks['cache']
+    core_device = TestRun.disks['core']
+
+    cache_device.create_partitions([Size(20, Unit.GibiByte)])
+    core_device.create_partitions([Size(20, Unit.GibiByte)])
+
+    return cache_device.partitions, core_device.partitions
+
+
+def run_fio(file_size):
+    (Fio().create_command()
+        .target(io_target)
+        .read_write(ReadWrite.randwrite)
+        .io_engine(IoEngine.libaio)
+        .io_depth(16)
+        .file_size(file_size)
```

Comment:
Fio's `file_size` method receives parameter of type `Size`. You use `int` in your code.

Reply:
it worked 🙈 but good catch, I will fix that, thanks!:)